docs/server-elasticsearch-setup.md

```diff
@@ -0,0 +1,81 @@
+---
+id: server-elasticsearch-setup
+title: How to set up Elasticsearch for the Temporal Server
+sidebar_label: Set up Elasticsearch
+---
+
+## Overview
+
+You can integrate [Elasticsearch](https://www.elastic.co/elasticsearch/) with the Temporal Server to enhance Workflow search functionality.
+The [Server versions and dependencies](/docs/server-versions-and-dependencies) page lists compatible versions of dependencies.
+To set up Elasticsearch, there are two main Server configurations that need to be edited.
+
+- `persistence`
+- `kafka`
+
+Note that you can use the Temporal Server [development_es.yaml](https://github.com/temporalio/temporal/blob/master/config/development_es.yaml) file as working example.
+
+## Persistence
+
+Elasticsearch relies on storage to operate.
+Storage related configurations are defined within the `persistence` section of the configuration file.
+The Temporal Server needs to know where to connect to the Elasticsearch storage.
+
+To do this, first add the `advancedVisibilityStore: es-visibility` key value pair to the `persistence` section.
+
+```yaml
+persistence:
+  ...
+  advancedVisibilityStore: es-visibility
+```
+
+Next, define the ElasticSearch datastore connection information under the `es-visibility` key:
+
+```yaml
+persistence:
+  ...
+  datastores:
+    ...
+    es-visibility:
+      elasticsearch:
+        version: "v7"
```

Comment:
@alexshtin is v7 ready?

Reply:
yes, I would leave it as "v7" here. Probably worth to mention that "v6" is also supported (although we going to remove it in near future).

Reply:
I have added this in the versions and dependencies page as well.